tests/trainer/logging_/test_eval_loop_logging.py

```diff
@@ -770,3 +773,155 @@ def test_dataloader(self):
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+┃       Test metric       ┃      DataLoader 0       ┃       DataLoader 1       ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+│          log1           │            5            │                          │
+│          log2           │            3            │                          │
+│         no_log1         │                         │            6             │
+│         no_log2         │                         │            1             │
+└─────────────────────────┴─────────────────────────┴──────────────────────────┘
+"""
+
+expected2 = """
+┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+┃           Validate metric            ┃             DataLoader 0              ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+│ a really really really really really │                   5                   │
+│  really really really really really  │                                       │
+│       really long metric name        │                                       │
+└──────────────────────────────────────┴───────────────────────────────────────┘
+┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+┃           Validate metric            ┃             DataLoader 1              ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+│ a really really really really really │                   6                   │
+│  really really really really really  │                                       │
+│       really long metric name        │                                       │
+└──────────────────────────────────────┴───────────────────────────────────────┘
+"""
+
+expected3 = """
+┏━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃   Foobar metric   ┃   DataLoader 0    ┃   DataLoader 1    ┃   DataLoader 2   ┃
+┡━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│        log        │         5         │         5         │        5         │
+└───────────────────┴───────────────────┴───────────────────┴──────────────────┘
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+┃      Foobar metric      ┃      DataLoader 3       ┃       DataLoader 4       ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+│           log           │            5            │            5             │
+└─────────────────────────┴─────────────────────────┴──────────────────────────┘
+"""
+
+
+@pytest.mark.parametrize(
+    ["inputs", "expected"],
+    [
+        pytest.param(inputs0, expected0, id="case0"),
+        pytest.param(inputs1, expected1, id="case1"),
+        pytest.param(inputs2, expected2, id="case2"),
+        pytest.param(inputs3, expected3, id="case3"),
+    ],
+)
+@RunIf(rich=True)
```

Comment:
@AndresAlgaba or @akihironitta, can you look into the Windows failures?
We might need to replace a character with another in the tests or just skip them for windows.

Reply:
I believe the problem is in the "\r\n vs \n newlines in Python on Mac vs Windows": https://stackoverflow.com/questions/4599936/handling-r-n-vs-n-newlines-in-python-on-mac-vs-windows. What is the preferred solution in this case?

Reply:
Would doing

`assert out.getvalue().replace(os.linesep, "\n") == expected.lstrip()`

work?

Reply:
Cool! I believe this works. Pushed a commit :)

Reply:
`rich` tests need it too